pkg/asset/agent/manifests/agentpullsecret.go

```diff
@@ -112,11 +115,17 @@ func (a *AgentPullSecret) finish() error {
 	}
 
 	// Normalise the JSON formatting so that we can redact the file reliably
-	normal, err := normalizeDockerConfig(a.Config.StringData[pullSecretKey])
+	pullSecretValue, ok := a.Config.StringData[pullSecretKey]
+	if !ok {
+		pullSecretValue = string(a.Config.Data[pullSecretKey])
```

Comment:
Need to decode from base64 here. Can we call a.GetPullSecretData() to avoid duplicating the implementation?